### PR TITLE
Changing animation state

### PR DESCRIPTION
**Pull request by JorisWillig**

### CODE REVIEW

File: game/systems/PlayerAnimationSystem.h [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+#pragma once
+
+#include <engine/ecs/World.h>
+#include <game/components/SpriteComponent.h>
+
+namespace game {
+namespace systems {
+    class PlayerAnimationSystem : public engine::ecs::BaseSystem<PlayerAnimationSystem> {
+    private:
+        engine::ecs::World& m_ecsWorld;
```
Comment (notzain) [inline game/systems/PlayerAnimationSystem.h]: pointer

File: game/systems/PlayerAnimationSystem.h [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#include <utility>
```
Comment (notzain) [inline game/systems/PlayerAnimationSystem.h]: under #pragma once

File: engine/graphics/drawable/Sprite.cpp [hunk 3]
```diff
@@ -69,6 +70,21 @@ void Sprite::setSize(const common::Vector2D<int>& size)
     m_size = size;
 }
 
+void Sprite::setFlippedVert(bool flipped)
```
Comment (notzain) [inline engine/graphics/drawable/Sprite.cpp]: prefer isFlipped. Also prefer not shortening the axis, so just write isFlippedHorizontal/Vertical

File: engine/graphics/drawable/Sprite.cpp [hunk 4]
```diff
@@ -84,4 +100,19 @@ void Sprite::draw(const IRenderer& renderer) const
     renderer.draw(*this);
 }
 
+bool Sprite::getFlippedVert() const
```
Comment (notzain) [inline engine/graphics/drawable/Sprite.cpp]: also.. prefer isFlipped

File: engine/graphics/drawable/Sprite.cpp (hunk 3 above)
Comment (JorisWillig) [reply engine/graphics/drawable/Sprite.cpp]: For the getter isFlipped is fine, but I think that would be weird for the setter.

File: engine/graphics/SDL/SDLRenderVisitor.cpp [hunk 5]
```diff
@@ -171,7 +171,8 @@ void SDLRenderVisitor::visit(const Sprite& sprite) const
         actualSourceSize.y
     };
 
-    SDL_RenderCopy(renderer, texture, &sourceRect, &positionRect);
+    SDL_RendererFlip rendererFlip = sprite.isFlippedHorizontal() ? SDL_FLIP_HORIZONTAL : SDL_FLIP_NONE;
```
Comment (FabioGNR) [inline engine/graphics/SDL/SDLRenderVisitor.cpp]: Add vertical flip support

File: engine/graphics/drawable/Sprite.cpp [hunk 6]
```diff
@@ -69,6 +70,21 @@ void Sprite::setSize(const common::Vector2D<int>& size)
...
 }
 
+void Sprite::setFlippedVertical(bool isFlipped)
+{
+    m_flippedVert = isFlipped;
+}
+
+void Sprite::setFlippedHorizontal(bool isFlipped)
+{
+    m_flippedHor = isFlipped;
```
Comment (FabioGNR) [inline engine/graphics/drawable/Sprite.cpp]: Hor to horizontal

File: game/CMakeLists.txt [hunk 7]
```diff
@@ -104,7 +105,7 @@ set(DEFINITION_SRC
         definitions/WeaponType.h
         definitions/ProjectileType.h
         definitions/ItemType.h
-        )
+        definitions/CharacterState.h)
```
Comment (FabioGNR) [inline game/CMakeLists.txt]: newline

File: game/CMakeLists.txt [hunk 8]
```diff
@@ -171,7 +172,8 @@ set(BUILDER_SRC
         sprite/SpriteDomain.h
         builders/SpriteBuilder.cpp
         builders/SpriteBuilder.h
-        )
+        systems/PlayerAnimationSystem.cpp
+        systems/PlayerAnimationSystem.h)
```
Comment (FabioGNR) [inline game/CMakeLists.txt]: newline

File: game/builders/CharacterBuilder.cpp [hunk 9]
```diff
@@ -146,6 +149,9 @@ namespace builders {
 
             // Remove the position
             positions.erase(positions.begin() + randomValue - 1);
+            entityAnimations.insert({ players[i].get(), playerAnimations[i] });
+
+            m_ecsWorld.addSystem<systems::PlayerAnimationSystem>(engine::definitions::SystemPriority::Medium, &m_ecsWorld, entityAnimations);
```
Comment (FabioGNR) [inline game/builders/CharacterBuilder.cpp]: move system out of loop, maybe to GameState?

**Pull request merged**